From 689ec94d8b3b2cd86ad58f780501fd6631c6d57f Mon Sep 17 00:00:00 2001
From: Peter Harliman Liem <peter.harliman.liem@intel.com>
Date: Wed, 15 Aug 2018 16:28:56 +0800
Subject: [PATCH] DRVLIB_SW-876 - Fix ECC strength for Micron NAND flash

Micron use 4-bits/sector if OOB is 64B, 8-bits/sector if
OOB is 128B.
---
 drivers/mtd/ltq-spinand/ltq_spinand.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/ltq-spinand/ltq_spinand.c b/drivers/mtd/ltq-spinand/ltq_spinand.c
index b199ff6e764a..abaa3fbbf79e 100644
--- a/drivers/mtd/ltq-spinand/ltq_spinand.c
+++ b/drivers/mtd/ltq-spinand/ltq_spinand.c
@@ -227,7 +227,8 @@ static int spi_nand_manufacture_init(struct mtd_info *mtd, struct nand_chip *chi
 //		if (mtd->oobsize== 64)
 //			chip->ecc.layout = &micron_ecc_layout_64;
 
-		chip->ecc.strength = 4; /* 4 bits correction per page region (512 bytes) */
+		/* 4 bits/sector for 64 Bytes OOB, doubled for 128 Bytes OOB */
+		chip->ecc.strength = mtd->oobsize / 16;
 		chip->ecc.size     = mtd->writesize >> 2;
 		//chip->ecc.steps    = 4; /* 4 regions within page  NOTE: removed here since it is calculated in nand_scan_tail */
 		chip->ecc.bytes    = 8; /* pretty sure it is 8 bytes ECC per sub page from oob layout */
